0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,911 @@
+vs. the roles defined by its parent type. However, the relation type being subtyped must be abstract. 
+It is not possible to override roles inherited by concrete relation types.
+
+<!-- test-ignore -->
+```typeql
+define
+
+membership sub relation, abstract, relates parent, relates member;
+
+collection-membership sub membership, 
+relates parent-collection as parent, 
+relates collection-member as member;
+```
+
+In the example above, the `collection-membership` relation type subtypes the `membership` relation type, and overrides 
+the inherited `parent` role as `parent-collection` and the inherited `member` role as `collection-member`.
+
+<div class="note">
+[Note]
+The two examples above can be run back to back. The second one will update the `membership` relation type to make it 
+abstract, and the `collection-membership` type to override its inherited roles.
+</div>
+
+##### Complex example
+
+<!-- test-ignore -->
+```typeql
+define
+
+ownership-type sub attribute, value string;
+
+ownership sub relation, abstract, 
+relates owned, 
+relates owner;
+
+group-ownership sub ownership, 
+relates owned-group as owned, 
+relates group-owner as owner, 
+owns ownership-type;
+
+object-ownership sub ownership, 
+relates owned-object as owned, 
+relates object-owner as owner, 
+owns ownership-type;
+
+access sub relation, 
+relates accessed-object, 
+relates valid-action, 
+plays change-request:requested-change;
+
+change-request sub relation, 
+relates requesting-subject, 
+relates requested-subject, 
+relates requested-change;
+```
+
+The example above defines one attribute type and five relation types:
+
+* `ownership` — subtypes the base `relation` type, is abstract, and relates `owned` and `owner` roles.
+* `group-ownership` — subtypes `ownership` relation type, relates `owned-group` as `owned` and `group-owner` as `owner`.
+* `object-ownership` — subtypes `ownership` relation, relates `owned-object` as `owned` and `object-owner` as `owner`.
+* `access` — subtypes the `relation` base type, relates `accessed-object` (e.g., files) and `valid-actions` (e.g., 
+  read), plays the role of `requested-change` in `change-request` relation type.
+* `change-request` — subtypes the `relation` base type, relates `requesting-subject`, `requested-subject` and
+  `requested-change`.
+
+### Define rules
+
+#### Overview
+
+Rules are defined independently of any types.
+
+They are executed only as a part of get queries when the [inference]() option is **enabled**. The results of rules 
+execution exist only within the transaction they run in. They are not persisted, and any data inferred in the 
+transaction ceases to exist when the transaction is closed. Read more on rules in the [Rules](#rules) section.
+
+#### Syntax
+
+Rules are defined in TypeQL with the following syntax:
+
+<!-- test-ignore -->
+```typeql
+rule <label>:
+when {<pattern>  [(, <pattern>)...]}
+then <pattern>
```

Comment:
`then` requires braces around the pattern. 
Patterns aren't comma-separated.

The way parentheses and brackets are used in these syntax references, I wouldn't be surprised if people get confused here. Suggestion:
```suggestion
rule <label>:
when {
    <pattern>  [(<pattern>)...]
} then {
    <pattern>
};
```